### PR TITLE
fix: alipay login garbled text

### PR DESCRIPTION
**Pull request by wanghaolonggit**

### CODE REVIEW

File: idp/alipay.go [hunk 1]
```diff
@@ -204,17 +207,23 @@ func (idp *AlipayIdProvider) postWithBody(body interface{}, targetUrl string) ([
...
-	}
 	defer func(Body io.ReadCloser) {
 		err := Body.Close()
 		if err != nil {
 			return
 		}
 	}(resp.Body)
 
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New("http status code is not 200")
```
Comment (hsluoyz) [inline idp/alipay.go:218]: Show error details, like status code, error message if has

Use fmt.Errorf

File: idp/alipay.go [hunk 2]
```diff
@@ -204,17 +207,23 @@ func (idp *AlipayIdProvider) postWithBody(body interface{}, targetUrl string) ([
...
+	}
+
+	utf8Reader := transform.NewReader(resp.Body,
+		simplifiedchinese.GBK.NewDecoder())
```
Comment (hsluoyz) [inline idp/alipay.go:222]: Do you have to import a 3rd-party lib to do this?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: idp/alipay.go (hunk 1 above)
Comment (wanghaolonggit) [reply idp/alipay.go:218]: ok

File: idp/alipay.go (hunk 2 above)
Comment (wanghaolonggit) [reply idp/alipay.go:222]: > Do you have to import a 3rd-party lib to do this?


I'll try to implement it myself

Comment (dacongda) [reply idp/alipay.go:222]: what will happen if  the nickName of alipay user is other language like japanese、korean ...

Comment (dacongda) [reply idp/alipay.go:222]: and it seems that we have set chatset to be utf-8 in 
https://github.com/casdoor/casdoor/blob/e99b56620be604a1a1dbe3d455a893d4f27d3040/idp/alipay.go#L102

Comment (wanghaolonggit) [reply idp/alipay.go:222]: > what will happen if the nickName of alipay user is other language like japanese、korean ...

all use gbk, after conversion, all  language  is normal.

Comment (wanghaolonggit) [reply idp/alipay.go:222]: > Do you have to import a 3rd-party lib to do this?

I think 3rd-party lib is more mature and robust

Comment (dacongda) [reply idp/alipay.go:222]: > > what will happen if the nickName of alipay user is other language like japanese、korean ...
> 
> all use gbk, after conversion, all language is normal.

can we fix bug by set line 102 from `utf-8` to `UTF-8` ?

Comment (wanghaolonggit) [reply idp/alipay.go:222]: > > > what will happen if the nickName of alipay user is other language like japanese、korean ...
> > 
> > 
> > all use gbk, after conversion, all language is normal.
> 
> can we fix bug by set line 102 from `utf-8` to `UTF-8` ?

yes

File: go.mod [hunk 3]
```diff
@@ -63,6 +63,7 @@ require (
 	golang.org/x/crypto v0.21.0
 	golang.org/x/net v0.21.0
 	golang.org/x/oauth2 v0.17.0
+	golang.org/x/text v0.14.0
```
Comment (hsluoyz) [inline go.mod:66]: Update go.sum

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: idp/alipay.go (hunk 2 above)
Comment (wanghaolonggit) [reply idp/alipay.go:222]: > and it seems that we have set charset to be utf-8 in
> 
> https://github.com/casdoor/casdoor/blob/e99b56620be604a1a1dbe3d455a893d4f27d3040/idp/alipay.go#L102

encoding format for  request

File: idp/alipay.go [hunk 4]
```diff
@@ -204,17 +207,23 @@ func (idp *AlipayIdProvider) postWithBody(body interface{}, targetUrl string) ([
 	if err != nil {
```
Comment (dacongda) [inline idp/alipay.go:207]: set this line to `resp, err := idp.Client.Post(targetUrl, "application/x-www-form-urlencoded;charset=utf-8", strings.NewReader(formData.Encode()))` can solve this issue without add extra lib

Comment (dacongda) [reply idp/alipay.go:207]: can you help me to test ?

Comment (dacongda) [reply idp/alipay.go:207]: @wanghaolonggit have you seen this info , can you help me to test?

**Comment by hsluoyz** [comment]
Replaced by: https://github.com/casdoor/casdoor/pull/3247

**Pull request closed without merging**